### PR TITLE
Docs for cf_units

### PR DESCRIPTION
**Pull request by DPeterK**
cf_units has no docs :scream: 

This PR rectifies the situation by adding a `doc` folder containing some docs generated with a sphinx quickstart and updated to document `cf_units.Unit` and the key utility functions.

This PR should allow us to hook into readthedocs and host the cf_units docs on there.

### CODE REVIEW

File: doc/source/conf.py [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+copyright = 'British Crown Copyright 2015 - 2016, Met Office'
+
+# The version info for the project you're documenting, acts as replacement for
+# |version| and |release|, also used in various other places throughout the
+# built documents.
+#
+# The short X.Y version.
+version = '1.0'
+# The full version, including alpha/beta/rc tags.
+release = '1.0.0'
```
Comment (pelson) [inline doc/source/conf.py]: Nice if this were correct.

**Comment by pelson** [comment]
:+1: nice start, thanks @dkillick.

A few nice additions to be considered:
- [ ] turn on RTD
- [x] Put the correct version in the docs
- [ ] Improve the template
- [ ] Test the doc building with travis, asserting that no warnings are generated, and links aren't dead
- [ ] Figure out a way to prevent public functions and classes from being undocumented (currently, we have remember to enumerate each item manually).

Other than the version number in the docs, the rest are nice-to-have in other PRs.

:+1:

File: doc/source/conf.py (hunk 1 above)
Comment (marqh) [inline doc/source/conf.py]: indeed; this library is using versioneer, so we need to hook into this to get the correct version

please may this be fixed?

**Comment by DPeterK** [comment]
The above commit makes use of `cf_units.__version__` to set the version and release strings in `doc/source/conf.py`. I couldn't get versioneer to import within `conf.py` so I figure using the versioneer-controlled string via the middle-man of `cf_units.__version__` isn't too bad a compromise.

**Comment by marqh** [comment]
this looks good to me

much obliged @dkillick

**Pull request merged**